### ISSUE TITLE
Testing

### ISSUE THREAD

**Issue by ispringle** [plusuncold/epub-bouncer#3]
Opening this issue to discuss the best way of going about providing some level of automated testing, something that can be the bare minimum that needs to be passed prior to a merge. This issue's successful completion will require a few PRs, including #2. Other needed PRs will include something to get Python testing going and something for ci/cd (probably GH Actions?)

**Comment by drewpearce** [comment]
I use pytest for unit testing and currently tend toward Travis for automated testing (but that's mostly because Travis makes it easy to deploy to pypi, and to connect to GH.)

**Comment by plusuncold** [comment]
`pytest` and Travis seem good - considering GH integrations.

**Comment by ispringle** [comment]
I have found GH actions to be very usable. No experience with Travis, we use Drone at work.

**Comment by plusuncold** [comment]
OK, got `pytest` running - which spotted a couple of small bugs actually. The code coverage is low at the moment, but we can get working on that.

Also got Travis working too, which currently just points to `pytest` but I'll add `flake8` to this now.

**Comment by plusuncold** [comment]
Regarding docker I'd say we're best coming back to it once I have some non-copyrighted EPUBs to include in the test suite.

**Comment by ispringle** [comment]
Could always use the Bible. https://ebible.org/find/details.php?id=eng-web

**Comment by plusuncold** [comment]
Yes, that would be a good source - would be a good test of ignoring capitalised words too. I'll get another public domain book and also de-copyright some eBooks that I have that have errors.
